packages/lesswrong/lib/filterSettings.ts

```diff
@@ -9,9 +16,157 @@ export interface FilterTag {
+    terms: {
+      view: "suggestedFilterTags",
+    },
+    collectionName: "Tags",
+    fragmentName: "TagPreviewFragment",
+    limit: 100,
+  })
+  
+  if (suggestedTags) {
+    filterSettings = addSuggestedTagsToSettings(filterSettings, suggestedTags)
+  }
+  
+  /** Set the whole mess */
+  const setFilterSettings = useCallback((newSettings: FilterSettings) => {
+    setFilterSettingsLocally(newSettings)
+    void updateCurrentUser({
+      frontpageFilterSettings: newSettings,
+    })
+  }, [updateCurrentUser])
+  
+  const setPersonalBlogFilter = useCallback((mode: FilterMode) => {
+    setFilterSettings({
+      personalBlog: mode,
+      tags: filterSettings.tags,
+    })
+  }, [setFilterSettings, filterSettings])
+  
+  /** Upsert - tagName required for insert */
+  const setTagFilter = useCallback(({tagId, tagName, filterMode}: {tagId: string, tagName?: string, filterMode: FilterMode}) => {
+    // update
+    let existingTagFilter = filterSettings.tags.find(tag => tag.tagId === tagId)
+    if (existingTagFilter) {
+      const replacedIndex = findIndex(filterSettings.tags, t => t.tagId === tagId)
+      let newTagFilters = [...filterSettings.tags]
+      newTagFilters[replacedIndex] = {
+        ...filterSettings.tags[replacedIndex],
+        filterMode,
+      }
+      setFilterSettings({
+        personalBlog: filterSettings.personalBlog,
+        tags: newTagFilters,
+      })
+      captureEvent('tagFilterModified', {tagId, tagName, newMode: filterMode})
+      return
+    }
+    // insert
+    if (!tagName) {
+      throw new Error("tagName required for insert")
+    }
+    captureEvent("tagAddedToFilters", {tagId, tagName})
+    setFilterSettings({
+      personalBlog: filterSettings.personalBlog,
+      tags: [...filterSettings.tags, { tagId, tagName, filterMode }],
+    })
+  }, [setFilterSettings, filterSettings, captureEvent])
+  
+  const removeTagFilter = useCallback((tagId: string) => {
+    if (suggestedTags.find(tag => tag._id === tagId)) {
+      throw new Error("Can't remove suggested tag")
+    }
+    captureEvent("tagRemovedFromFilters", {tagId});
+    const newTags = filter(filterSettings.tags, tag => tag.tagId !== tagId)
+    setFilterSettings({
+      personalBlog: filterSettings.personalBlog,
+      tags: newTags,
+    })
+  }, [setFilterSettings, filterSettings, suggestedTags, captureEvent])
+  
+  return {
+    filterSettings,
+    loadingSuggestedTags,
+    errorLoadingSuggestedTags,
+    setFilterSettings,
+    setPersonalBlogFilter,
+    removeTagFilter,
+    setTagFilter,
+  }
+}
+
+/**
+ * A simple wrapper on top of useFilterSettings focused on a single tag
+ * subscription
+ */
+export const useSubscribeUserToTag = (tag: TagBasicInfo) => {
+  const { filterSettings, setTagFilter } = useFilterSettings()
+  
+  const filterSetting = filterSettings.tags.find(ft => ft.tagId === tag._id)
+  const isSubscribed = filterSetting?.filterMode === "Subscribed" || filterSetting?.filterMode === 25
```

Comment:
From above comment, if you decide to change 'Subscribed' to mean >=25, this line would need to change too.

Reply:
I do think this one should turn to subscribed for a filterMode >25.